.github/workflows/ci.yml

```diff
@@ -1,58 +1,47 @@
-name: CI
+name: Test and release when tagged
 
-on: [push, pull_request]
+on: [push]
 
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: prefix-dev/setup-pixi@v0.8.1
+        with:
+          environments: lint
+      - run: pixi run --environment lint lint
+
   test-linux:
+    runs-on: ubuntu-latest
     strategy:
-      fail-fast: false
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+        environment:
+          - test-py39
```

Comment:
You've removed support for 3.9 and 3.10 in pyproject.toml.

Maybe keep 3.10 for now according to https://numpy.org/neps/nep-0029-deprecation_policy.html
We can still support it until April 04 2025.

Reply:
Saw that you're using 3.11 only features though. Well I'mnot against of moving forward. Only thing might be some downstream that still support 3.10 like Metpy and others.

Reply:
The new Pint version will be released on > April 05, I am not sure if we should. Maybe we can list what would be the burden of keeping Python 3.10